pyscf/pbc/gto/cell.py

```diff
@@ -589,13 +589,21 @@ def get_Gv_weights(cell, mesh=None, **kwargs):
     rx = np.asarray(rx, order='C')
     ry = np.asarray(ry, order='C')
     rz = np.asarray(rz, order='C')
-    fn = libpbc.get_Gv
-    fn(Gv.ctypes.data_as(ctypes.c_void_p),
-       rx.ctypes.data_as(ctypes.c_void_p),
-       ry.ctypes.data_as(ctypes.c_void_p),
-       rz.ctypes.data_as(ctypes.c_void_p),
-       mesh.ctypes.data_as(ctypes.c_void_p),
-       b.ctypes.data_as(ctypes.c_void_p))
+
+    fn = getattr(libpbc, 'get_Gv', None)
+    assert fn is not None
+
+    try:
+        fn(Gv.ctypes.data_as(ctypes.c_void_p),
+        rx.ctypes.data_as(ctypes.c_void_p),
+        ry.ctypes.data_as(ctypes.c_void_p),
+        rz.ctypes.data_as(ctypes.c_void_p),
+        mesh.ctypes.data_as(ctypes.c_void_p),
+        b.ctypes.data_as(ctypes.c_void_p))
+
+    except Exception:
```

Comment:
Why calling the C function can fail? Can this error be more specific?

Reply:
I updated pyscf to master and didn't compile the new binary file. The code breaks in some weird place. Adding this assert might help people (like me) to locate the problem. It shall also be the common practice in pyscf when call C functions.

Reply:
If the symbol not available from the C extension, the code should be failed at the line `fn = getattr(libpbc, 'get_Gv', None)` . You can change it to `fn libpbc.get_Gv` , which will give an accurate error info. Catching errors for the C caller is misleading.